pkg/rtc/subscribedtrack.go

```diff
@@ -215,14 +215,14 @@ func (t *SubscribedTrack) UpdateVideoLayer() {
 		return
 	}
 
-	t.logger.Debugw("updating video layer",
-		"settings", settings,
-	)
-
-	spatial := t.spatialLayerFromSettings(settings)
-	t.DownTrack().SetMaxSpatialLayer(spatial)
-	if settings.Fps > 0 {
-		t.DownTrack().SetMaxTemporalLayer(t.MediaTrack().GetTemporalLayerForSpatialFps(spatial, settings.Fps, t.DownTrack().Codec().MimeType))
+	t.logger.Debugw("updating video layer", "settings", settings)
+
+	if settings.Width > 0 || settings.Fps > 0 {
+		spatial := t.spatialLayerFromSettings(settings)
+		t.DownTrack().SetMaxSpatialLayer(spatial)
+		if settings.Fps > 0 {
+			t.DownTrack().SetMaxTemporalLayer(t.MediaTrack().GetTemporalLayerForSpatialFps(spatial, settings.Fps, t.DownTrack().Codec().MimeType))
+		}
```

Comment:
Adding this check here to not set max spatial layer on mute. When `settings.disabled: true` happens, that is usually the only thing that is set. But, we process layer  also even if width/height is not set and end up setting maximum spatial layer to 0 (LOW quality). On an unmute, subscription notifies layer 0 as desired and then switches after the next video layer update. Avoiding that.